back-end/service.py

```diff
@@ -37,49 +47,75 @@ def userData(token):
 
 def ensureToken(username, password):
 
-    res = requests.get(
-        GIN_ADDR + "/api/v1/users/{}/tokens".format(username),
-        auth=(username, password)).json()
+    """
+    Retrieves the personal access token `gin-proc`
+    from user's GIN account to be used further in session.
+
+    In case, the specific token for gin-proc doesn't exists,
+    it registers a fresh token to GIN for that user.
+    """
 
-    for token in res:
-        if token['name'] == 'gin-proc':
-            return token['sha1']
+    try:
+        res = requests.get(
+            GIN_ADDR + "/api/v1/users/{}/tokens".format(username),
+            auth=(username, password)).json()
+
+        for token in res:
+            if token['name'] == 'gin-proc':
+                return token['sha1']
+
+        res = requests.post(
+            GIN_ADDR + "/api/v1/users/{}/tokens".format(username),
+            auth=(username, password),
+            data={'name': 'gin-proc'}
+        ).json()
 
-    res = requests.post(
-        GIN_ADDR + "/api/v1/users/{}/tokens".format(username),
-        auth=(username, password),
-        data={'name': 'gin-proc'}
-    ).json()
+        return res['sha1']
 
-    return res['sha1']
+    except Exception.ConnectionError as e:
+        raise ServerError(e)
 
 
 def writeSecret(key, repo, user):
 
-    res = requests.post(
-        DRONE_ADDR + "/api/repos/{0}/{1}/secrets".format(user, repo),
-        headers={
-            'Authorization': 'Bearer {}'.format(os.environ['DRONE_TOKEN']),
-            'Content-Type': "application/json"
-            },
-        json={
-            "name": "DRONE_PRIVATE_SSH_KEY",
-            "data": key,
-            "pull_request": False
-            }
-    )
+    """
+    Writes the key as a secret title `DRONE_PRIVATE_SSH_KEY`
+    to specified repository in Drone.
+    """
+
+    try:
+        res = requests.post(
+            DRONE_ADDR + "/api/repos/{0}/{1}/secrets".format(user, repo),
+            headers={
+                'Authorization': 'Bearer {}'.format(os.environ['DRONE_TOKEN']),
+                'Content-Type': "application/json"
+                },
+            json={
+                "name": "DRONE_PRIVATE_SSH_KEY",
+                "data": key,
+                "pull_request": False
+                }
+        )
+
+    except Exception.ConnectionError as e:
+        raise ServerError(e)
 
     if res.status_code == 200:
         log('debug', 'Secret installed in `{}`'.format(repo))
         return True
     else:
-        log('warning', 'Secret could not be installed in `{}`'.format(repo))
         log('critical', res.json()['message'])
-        return False
+        raise ServerError('Secret could not be installed in `{}`'.format(repo), 410)
```

Comment:
Is `410 - Gone` the appropriate http.StatusCode at this point?

It would also be nice to use `http.StatusCodes` instead of just numbers.

Reply:
Changed the status codes and used `HTTPStatus` in latest push. Please have a look.

Reply:
I should have been more specific, sorry. Please use explicit status codes instead of passing the numbers so its easier to read e.g. `HTTPStatus.BAD_REQUEST` for code `400`.